src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -55,24 +54,21 @@ public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant
         if (files.length != 0) {
             for (LogFile file: files) {
                 String fileHash = file.hashString();
+                boolean isModifiedAfterLastUpdatedFile =
```

Comment:
why do we care about the timestamp? What is the intention behind checking this?

Reply:
This is to filter out files that has been processed before, if they are modified before the lastUpdated file

Reply:
We shouldn't be relying on timestamps for that. We should use the file size. If the file is longer than what we've already read, then we need to read it more and upload the rest.

Reply:
You are referring the case that rotated files may also still be adding some contents, right? But currently we don`t have a logic to record the bytesRead for each file, we only have it for the current processing file. This will be a big change, we can leave it as future refactoring.